### PR TITLE
Fix Meta modules and add inputs to ndvi-colormap

### LINKED ISSUE #1239: problem with meta-modules
There are many errors occuring while using meta-module(colorbar and ndvi-colormap).
Steps to reproduce
1 .chose nvdi-colormap(or colorbar)
2.then chose any other module and change the value that is not the default value
it would now change image produced by the above mentioned module as well .

Screencastfrom29-08-19095222PMIS

****

## Thank you!

Your help makes Public Lab better! We *deeply* appreciate your helping refine and improve this site.

To learn how to write really great issues, which increases the chances they'll be resolved, see:

https://publiclab.org/wiki/developers#Contributing+for+non-coders

### LINKED ISSUE #1418: Image processing not correct in colorbar module
### Please describe the problem (or idea)
When name of the colormap is changed(doesnot matter from default to greyscale or greyscale to default), it doesnot work corectly.
> What happened just before the problem occurred? Or what problem could this idea solve?
bug

> What did you expect to see that you didn't?

The image for greyscale colobar on the top with 10px height.

### Please show us where to look

https://beta.sequencer.publiclab.org

### What's your PublicLab.org username?

> This can help us diagnose the issue:

### Browser, version, and operating system

> Many bugs are related to these -- please help us track it down and reproduce what you're seeing!

****

## Thank you!

Your help makes Public Lab better! We *deeply* appreciate your helping refine and improve this site.

To learn how to write really great issues, which increases the chances they'll be resolved, see:

https://publiclab.org/wiki/developers#Contributing+for+non-coders

### LINKED ISSUE #1431: Add input fields for NDVI and Colormap in ndvi-colormap module
### Please describe the problem (or idea)
As of now ndvi-colormap module doesn't support any input fields should we add input fields for NDVI and colormap?

### Please show us where to look

https://beta.sequencer.publiclab.org

### What's your PublicLab.org username?
rishabhshukla1999

> This can help us diagnose the issue:

### Browser, version, and operating system

> Many bugs are related to these -- please help us track it down and reproduce what you're seeing!

****

## Thank you!

Your help makes Public Lab better! We *deeply* appreciate your helping refine and improve this site.

To learn how to write really great issues, which increases the chances they'll be resolved, see:

https://publiclab.org/wiki/developers#Contributing+for+non-coders

### PR DESCRIPTION
**Pull request by blurry-x-face**
Fixes #1239 ,Fixes #1418 , Fixes #1431   

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `npm test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] ask `@publiclab/is-reviewers` for help, in a comment below
* [x] Insert-step functionality is working correct as expected.

Below is the working GIFS
meta1

meta2

### CODE REVIEW

**Comment by blurry-x-face** [comment]
@publiclab/is-reviewers @jywarren @keshav234156  please review this again, In my last PR whole bug was not resolved.

File: src/modules/NdviColormap/info.json [hunk 1]
```diff
@@ -1,6 +1,24 @@
 {
     "name": "ndvi-colormap",
     "description": "Sequentially Applies NDVI and Colormap steps",
-    "inputs": {},
+    "inputs": {
```
Comment (HarshKhandeparkar) [inline src/modules/NdviColormap/info.json]: Wow. That's a big fix. Very important.

File: src/util/createMetaModule.js [hunk 2]
```diff
@@ -40,7 +35,13 @@ module.exports = function createMetaModule(mapFunction, moduleOptions) {
     function draw(input, callback) {
 
       var step = this;
-      
+
+      // map inputs to internal step options;
+      // use this to set defaults for internal steps
+      // and to expose internal settings as external meta-module parameters;
+      // it must return a steps object
+      var steps = mapFunction(options);
```
Comment (jywarren) [inline src/util/createMetaModule.js]: Hi @tech4GT can you check this change of scope? Thank you!

**Comment by jywarren** [comment]
This is looking great; i asked @tech4GT to review as he most recently worked on these files. Would it be possible to confirm this change with a test? https://github.com/publiclab/image-sequencer/blob/main/test/core/modules/colormap.js

File: src/util/createMetaModule.js (hunk 2 above)
Comment (jywarren) [reply src/util/createMetaModule.js]: Linking to explanation: https://github.com/publiclab/image-sequencer/issues/1239#issuecomment-571337844

**Comment by keshav234156** [comment]
@jywarren after #1239 the problem that still remained was that when option where changes is still processed the image with original option. This thing we are currently not testing that why they always passed. To test this we need to change https://github.com/publiclab/image-sequencer/blob/main/test/core/templates/module-test.js and all test modules using it .It would be a big change

**Comment by blurry-x-face** [comment]
Yes, this is the reason why the current tests weren't able to detect it and we can make those changes in a new PR and I will be happy to make these changes but lets merge this one for now.

**Comment by jywarren** [comment]
OK, that sounds good. Just checking! Thank you!!!

**Pull request merged**